src/Microsoft.Diagnostics.Runtime/ClrValue.cs

```diff
@@ -0,0 +1,1077 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Runtime
+{
+    /// <summary>
+    /// A ClrValue represents the value of a field or variable.  This value may be of any type (or null).
+    /// </summary>
+    public abstract class ClrValue
```

Comment:
You might consider making an interface IClrValue which is implemented by both ClrObject and ClrValue? Alternatively if ClrObject was a reference type you could derive from ClrValue. In my mind if I wanted a really high performance object enumerator I'd just use the raw addresses. As soon as you are willing to pay the cost to look up the method table and create a reference for it, it doesn't feel like I am saving much having ClrObject be a struct. All that said, I have 0 perf data in front of me and this is wild speculation : )